### ISSUE TITLE
UI/UX suggestions

### ISSUE THREAD

**Issue by dhirschfeld** [lckr/jupyterlab-variableInspector#51]
First, thanks for working on this extension - I think it's incredibly important functionality for JupyterLab and is one of the most requested features from my users!

I'm just starting to try it out and have a couple of suggestions.

![image](https://user-images.githubusercontent.com/881019/45602510-0ef0f080-ba63-11e8-85a8-9e41e1b3991e.png)

 * Since space is at a premium the shape column should just show the shape tuple - e.g. `(100, 3)`.
   The datatype (Array/DataFrame) is redundant since it just duplicates the TYPE column
   The rows/columns are redundant since they're implicit in the ordering
 * After the type and shape the next most important information for me is the dtype so I'm wondering if there is some way to incorporate that? For the DataFrame seeing the column names (under content) is nice but it woudl be more useful if it were a `name: type` mapping. I can imagine that might quickly become too much information to display but thought I'd mention it in case anyone has a good solution.

**Comment by lckr** [comment]
Hey @dhirschfeld,
Thank you for your valuable input.

 I agree with you that there is too much redundant information in the type and shape fields at the moment. 

Additionally, I like the idea of showing `dtypes` in the content field. But as you already pointed out, it is not too easy to decide what to show in this field due to space constraints. I'm not very happy with the current state especially w.r.t. the presentation of arrays which I find quite ugly as soon as its string representation needs to be truncated.

My idea is to show values only for scalars, strings and small collections. For larger collections or matrices we could try to display the 'dtypes'. To cope with the limited space we could allow to expand or collapse a row vertically whenever the enumeration of collums would exeed the available space.

What do you think about that?

**Comment by dhirschfeld** [comment]
Yeah, I think figuring out what/how to display an object is a hard problem!

I like your idea about an expander and had similar thoughts myself. A DataFrame may have come from a database with 20+ columns all with long names so even the dtype repr would have to be truncated. An expander might well be the solution there though.

**Comment by nmichaud** [comment]
Chrome's debugger has a good UX for lists/arrays. It will chunk up rows of data into bigger groups (say 1000 rows) and let you drill into those individual chunks.

**Comment by dhirschfeld** [comment]
***xref:*** #92
